tests/langchain/test_langchain_model_export.py

```diff
@@ -380,6 +382,111 @@ def load_retriever(persist_directory):
+
+    # Create the RetrieverWrapper
+    retriever_wrapper = RetrieverWrapper(retriever=db.as_retriever())
+
+    # Log the RetrieverWrapper
+    def load_retriever(persist_directory):
+        # pylint: disable=lazy-builtin-import
+        from typing import List
+
+        # pylint: enable=lazy-builtin-import
+        import numpy as np
+        from langchain.embeddings.base import Embeddings
+        from pydantic import BaseModel
+
+        class DeterministicDummyEmbeddings(Embeddings, BaseModel):
+            size: int
+
+            def _get_embedding(self, text: str) -> List[float]:
+                if isinstance(text, np.ndarray):
+                    text = text.item()
+                seed = abs(hash(text)) % (10**8)
+                np.random.seed(seed)
+                return list(np.random.normal(size=self.size))
+
+            def embed_documents(self, texts: List[str]) -> List[List[float]]:
+                return [self._get_embedding(t) for t in texts]
+
+            def embed_query(self, text: str) -> List[float]:
+                return self._get_embedding(text)
+
+        embeddings = DeterministicDummyEmbeddings(size=5)
+        vectorstore = FAISS.load_local(persist_directory, embeddings)
+        return vectorstore.as_retriever()
+
+    with mlflow.start_run():
+        logged_model = mlflow.langchain.log_model(
+            retriever_wrapper,
+            "retrieval_qa_chain",
+            loader_fn=load_retriever,
+            persist_dir=persist_dir,
+        )
+
+    # Remove the persist_dir
+    shutil.rmtree(persist_dir)
+
+    # Load the RetrieverWrapper
+    loaded_model = mlflow.langchain.load_model(logged_model.model_uri)
+    assert loaded_model == retriever_wrapper
+
+    loaded_pyfunc_model = mlflow.pyfunc.load_model(logged_model.model_uri)
+    query = "What did the president say about Ketanji Brown Jackson"
+    langchain_input = {"query": query}
+    result = loaded_pyfunc_model.predict([langchain_input])
+    expected_result = json.dumps(
+        [doc.page_content for doc in db.as_retriever().get_relevant_documents(query)]
+    )
+    assert type(result) == list
+    assert len(result) == 1
+    assert result[0] == expected_result
+
+    # Serve the chain
+    inference_payload = json.dumps({"inputs": langchain_input})
+    langchain_output_serving = {"predictions": [expected_result]}
+    response = pyfunc_serve_and_score_model(
+        logged_model.model_uri,
+        data=inference_payload,
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON,
+        extra_args=["--env-manager", "local"],
+    )
+    pred = PredictionsResponse.from_json(response.content.decode("utf-8"))["predictions"]
+    assert type(pred) == list
+    assert len(pred) == 1
+    docs_list = json.loads(pred[0])
+    assert type(docs_list) == list
+    assert len(docs_list) == 4
+    # The returned docs are non-deterministic when used with dummy embeddings
+    assert pred != langchain_output_serving
```

Comment:
Given we don't want to use a real embedding such as openai embeddings in our unit test, it's hard to implement a dummy embedding that returns deterministic doc collections. So I just check the shape of the prediction response here.